lib/mint/http2.ex

```diff
@@ -1324,6 +1318,30 @@ defmodule Mint.HTTP2 do
     end)
   end
 
+  defp add_pseudo_headers(headers, conn, "CONNECT", _path) do
+    [
+      {":method", "CONNECT"},
+      {":authority", authority_pseudo_header(conn.scheme, conn.port, conn.hostname)}
+      | headers
+    ]
+  end
+
+  defp add_pseudo_headers(headers, conn, method, path) do
+    [
+      {":method", method},
+      {":path", path},
+      {":scheme", conn.scheme},
+      {":authority", authority_pseudo_header(conn.scheme, conn.port, conn.hostname)}
+      | headers
+    ]
+  end
```

Comment:
We don't do any validation on the method being all upcase IIRC, so we need to check for that down below I think:

```suggestion
  defp add_pseudo_headers(headers, conn, method, path) do
    if String.upcase(method) == "CONNECT" do
      [
        {":method", method},
        {":authority", authority_pseudo_header(conn.scheme, conn.port, conn.hostname)}
        | headers
      ]
    else
      [
        {":method", method},
        {":path", path},
        {":scheme", conn.scheme},
        {":authority", authority_pseudo_header(conn.scheme, conn.port, conn.hostname)}
        | headers
      ]
  end
```